### ISSUE TITLE
WSL2 doesn't work, WSL1 does

### ISSUE THREAD

**Issue by ArtemVorobyev** [microsoft/WSL#8698]
### Version

Microsoft Windows [Version 10.0.19044.1865]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

Ubuntu 20.04

### Other Software

Docker Desktop

### Repro Steps

I'm using WSL on Windows 10, now I need  to install Docker Desktop.
Switching to WSL2 is somehow impossible.
Install WSL using PS wsl --install
Update WSL Kernel
Allow virtualization in BIOS, switch on VM and Windows Subsystem for Linux in features
Download and install Ubuntu-20.04 LTS and Docker Desktop, that failes if wsl --set-default-version 2, Ubuntu can be installed and functions only under wsl --set-default-version 1

### Expected Behavior

Switch to WSL2 and work further with Ubuntu and Docker Desktop

### Actual Behavior

If I set in PS wsl --set-default-version 2, then Ubuntu either doesn't install (giving the error 0x80370102), however, the virtualization was enabled in BIOS. 
The Ubuntu installation works if I switch back to PS wsl --set-default-version 1. 
If I try to set 
```
PS C:\Windows\system32> wsl --set-version Ubuntu-20.04 2
Conversion in progress, this may take a few minutes...
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
An address incompatible with the requested protocol was used.
```
Hyper-V and hypervisor platform are disabled in the features, VM and Windows Subsystem for Linux are enabled. I tried to enable Hyper-V and hypervisor, but it did not change anything.
So I suppose I have some problem with WSL2, but I don't know anymore where to look for it...
When I install Docker Desktop (tried different versions, with WSL2 and with hyper-V) - they don't start. Docker Desktop runs for several seconds and then switches off.

**Comment by NickDeBeenAIE** [comment]
It appears that your PC may not be compatible with WSL 2 and can only use WSL 1.

**Comment by ArtemVorobyev** [comment]
I finally solved the problem, but the solution was not as elegant as I was hoping.
I updated windows 10 to windows 11 and after that WSL2 and Docker Desktop are working perfectly.

While looking for the error still on windows10 using msinfo32, I noted that one driver for Hyper-V was showing some failure, so I was hoping that after reinstallation of the driver everything will be fine (as I have seen in some treads), but after I deleted this driver and reinstalled it I could not see any error message in msinfo32, but WSL2 was still not working. After that I updated my windows 10 to windows11 and everything started working.
